### ISSUE TITLE
Remove unwanted plugins from babel-config when compiling in esm mode

### ISSUE THREAD

**Issue by prateekbh** [developit/preact-cli#660]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behaviour?**
When compiling in ESM mode, we do not remove unwanted babel plugins.

**If the current behaviour is a bug, please provide the steps to reproduce.**
I have not tested this, but since I added the support for ESM I did not do any steps no avoid the following

Compile with `--esm` and check "fast-async" is still there?

**What is the expected behaviour?**
Plugins: 
https://github.com/developit/preact-cli/blob/next/packages/cli/lib/lib/babel-config.js#L20

https://github.com/developit/preact-cli/blob/next/packages/cli/lib/lib/babel-config.js#L23

https://github.com/developit/preact-cli/blob/next/packages/cli/lib/lib/babel-config.js#L27

Should not be present in ESM bundle

**If this is a feature request, what is motivation or use case for changing the behaviour?**
ESModules

**Please mention other relevant information.**
- CLI version: next

**Comment by prateekbh** [comment]
event async await?

**Comment by ForsakenHarmony** [comment]
we don't want the generator async/await from babel

**Comment by lukeed** [comment]
Sorry I looked at `object-assign` only & extrapolated the rest.

We should remove `object-assign` – Babel does this by default.

Anything with a `"proposal"` in its name is _not_ done by `@babel/preset-env` by default. I'd argue those have to/should stay when compiling for modules. It's still a matter of browser compatibility.

Fast-async should stay too. I'm not sure off-hand how to make Babel leave `async` alone, but it's probably just a matter of installing a "syntax" plugin

**Comment by prateekbh** [comment]
yep, my idea however is that, when the babel-esm-plugin runs this again in ESM mode, the pre-run hook should remove fast-async because ESM doesnt need it
